contracts/v0.2/src/SubmissionProxy.sol

```diff
@@ -407,4 +391,48 @@ contract SubmissionProxy is Ownable {
     function getAllOracles() public view returns (address[] memory) {
         return oracles;
     }
+
+    /**
+     * @notice Verify the proof
+     * @param _feed The address of the feed
+     * @param _message The hash of the message
+     * @param _proofs The proofs
+     * @return `true` if the proof is valid, `false` otherwise
+     */
+    function verifyProof(address _feed, bytes32 _message, bytes[] memory _proofs) private view returns (bool) {
+        uint8 verifiedSignatures_ = 0;
+        uint8 lastIndex_ = 0;
+        uint256 oracleCount_ = oracles.length;
+
+        uint8 threshold_ = thresholds[_feed];
+        if (threshold_ == 0) {
+            threshold_ = defaultThreshold;
+        }
+        uint8 requiredSignatures_ = quorum(threshold_);
+
+        uint256 proofsLength_ = _proofs.length;
+        for (uint256 j = 0; j < proofsLength_; j++) {
+            bytes memory proof_ = _proofs[j];
+            address signer_ = recoverSigner(_message, proof_);
+            uint8 oracleIndex_ = whitelist[signer_].index;
+
+            if (j != 0 && oracleIndex_ <= lastIndex_) {
+                revert IndexesNotAscending();
+            }
+            lastIndex_ = oracleIndex_;
+
+            if (oracleIndex_ >= oracleCount_) {
+                revert IndexOutOfBounds();
+            }
```

Comment:
wouldn't this case be caught from

```solidity
if (j != 0 && oracleIndex_ <= lastIndex_) {
                revert IndexesNotAscending();
            }
```

if so, I think it'd be okay to remove this check and error.

Reply:
It wouldn't be caught by the condition above, however, we might assume that the `index` value of `OracleInfo` struct is correctly (within bounds) set. I will remove the check.

Reply:
removed at https://github.com/Bisonai/orakl/pull/1410/commits/8d0b594113e49d9b11bbf521fb820f4ef98d97ba